batchpr/updater.py

```diff
@@ -232,19 +232,19 @@ def clone_fork(self, dirname='.'):
         self.run_command(f'git clone --depth 1 {self.fork.html_url}')
         os.chdir(self.repo.name)
 
-        # Make sure the branch doesn't already exist
         try:
-            self.run_command(f'git checkout origin/{self.branch_name}')
-        except:  # noqa
+            self.run_command("git remote set-branches origin '*'")
+            self.run_command(f'git fetch origin {self.branch_name}')
+        except Exception:  # noqa
             pass
-        else:
-            raise BranchExistsException()
 
-        # Update to the latest upstream's default branch (usually "master")
-        self.run_command(f'git remote add upstream {self.repo.html_url}')
-        self.run_command('git fetch upstream')
-        self.run_command(f'git checkout upstream/{self.repo.default_branch}')
-        self.run_command(f'git checkout -b {self.branch_name}')
+        try:
+            self.run_command(f'git checkout -b {self.branch_name} origin/{self.branch_name}')
+        except Exception:  # noqa
```

Comment:
I don't love that most of the runs will actually hit Exception block but this isn't my repo.